libyul/optimiser/StackCompressor.cpp

```diff
@@ -0,0 +1,107 @@
+/*(
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Optimisation stage that aggressively rematerializes certain variables ina a function to free
+ * space on the stack until it is compilable.
+ */
+
+#include <libyul/optimiser/StackCompressor.h>
+
+#include <libyul/optimiser/SSAValueTracker.h>
+#include <libyul/optimiser/NameCollector.h>
+#include <libyul/optimiser/Rematerialiser.h>
+#include <libyul/optimiser/UnusedPruner.h>
+#include <libyul/optimiser/Metrics.h>
+#include <libyul/optimiser/Semantics.h>
+
+#include <libyul/CompilabilityChecker.h>
+
+#include <libyul/AsmData.h>
+
+using namespace std;
+using namespace dev;
+using namespace yul;
+
+namespace
+{
+
+template <typename ASTNode>
+void eliminateVariables(shared_ptr<Dialect> const& _dialect, ASTNode& _node, size_t _numVariables)
+{
+	SSAValueTracker ssaValues;
+	ssaValues(_node);
+
+	map<YulString, size_t> references = ReferencesCounter::countReferences(_node);
+
+	set<pair<size_t, YulString>> rematCosts;
+	for (auto const& ssa: ssaValues.values())
+	{
+		if (!MovableChecker{*_dialect, *ssa.second}.movable())
+			continue;
+		size_t numRef = references[ssa.first];
+		yulAssert(numRef >= 0, "SSA variable with negative references.");
```

Comment:
This is superfluous (``size_t`` is unsigned) and as thus causes the clang build failure on travis.